### PR TITLE
tests: api: convert to dir helpers

### PR DESCRIPTION
**Pull request by pared**
* [x] ❗ Have you followed the guidelines in the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) list?

* [x] 📖 Check this box if this PR **does not** require [documentation](https://dvc.org/doc) updates, or if it does **and** you have created a separate PR in [dvc.org](https://github.com/iterative/dvc.org) with such updates (or at least opened an issue about it in that repo). Please link below to your PR (or issue) in the [dvc.org](https://github.com/iterative/dvc.org) repo.

* [x] ❌ Have you checked DeepSource, CodeClimate, and other sanity checks below? We consider their findings recommendatory and don't expect everything to be addressed. Please review them carefully and fix those that actually improve code or fix bugs.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: tests/func/test_api.py [hunk 1]
```diff
@@ -101,33 +102,34 @@ def _set_remote_url_and_commit(repo, remote_url):
...
+    erepo_dir.scm_gen({"scm_controlled": "file content"}, commit="create file")
 
-    stage_content = open(stage.path, "r").read()
-    with api.open(stage.path) as fd:
-        assert fd.read() == stage_content
+    with api.open("scm_controlled", repo=fspath(erepo_dir)) as fd:
+        assert fd.read() == "file content"
 
 
-# TODO: simplify, we shouldn't need run.
```
Comment (pared) [inline tests/func/test_api.py]: I don't think there is an easy way to have a non-cached file without `run`, an alternative could be `add`-ing and modifying stage file, but that is artificial, and probably will not require less work than `run`.

**Comment by pared** [comment]
Added WIP, Ill wait for https://github.com/iterative/dvc/pull/3043 to enter the master.

File: tests/func/test_api.py [hunk 2]
```diff
@@ -101,33 +102,34 @@ def _set_remote_url_and_commit(repo, remote_url):
...
+                    )
+                )
+            ),
+        )
+
+    with api.open("metric.txt", repo=fspath(erepo_dir)) as fd:
         assert fd.read() == metric_content
 
-    os.remove(metric_file)
+    os.remove(fspath(erepo_dir / "metric.txt"))
```
Comment (Suor) [inline tests/func/test_api.py]:
```suggestion
    (erepo_dir / "metric.txt").unlink()
```

File: tests/func/test_api.py [hunk 3]
```diff
@@ -101,33 +102,34 @@ def _set_remote_url_and_commit(repo, remote_url):
...
-    with api.open(stage.path) as fd:
-        assert fd.read() == stage_content
+    with api.open("scm_controlled", repo=fspath(erepo_dir)) as fd:
+        assert fd.read() == "file content"
 
 
-# TODO: simplify, we shouldn't need run.
-def test_open_not_cached(dvc):
-    metric_file = "metric.txt"
+def test_open_not_cached(dvc, erepo_dir, monkeypatch):
```
Comment (Suor) [inline tests/func/test_api.py]: No need to use `erepo_dir` here.

**Pull request merged**